### PR TITLE
Feature initial code

### PR DESCRIPTION
**Pull request by adrigzr**

### CODE REVIEW

File: tests/acceptance/chains-test.js [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+	await visit('/A/B');
+
+	assert.ok(routeA.onExit.notCalled, 'hook is not called');
+	assert.ok(routeAA.onExit.calledOnce, 'hook is called');
+});
+
+test('it does not call unnecesary hooks', async function(assert) {
+	const routeB = this.appInstance.lookup('route:B');
+	const routeAAA = this.appInstance.lookup('route:A.A.A');
```
Comment (josex2r) [inline tests/acceptance/chains-test.js]: It's like batteries O_O

File: addon/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+function didTransition() {
+	const routerMicrolib = this.router || this._routerMicrolib;
+	const onEnterChains = routerMicrolib.state.handlerInfos
+		.filter((info) => !routerMicrolib.oldState.handlerInfos.includes(info))
+		.map((info) => [info.handler, 'onEnter']);
+	const onExitChains = routerMicrolib.oldState.handlerInfos
+		.filter((info) => !routerMicrolib.state.handlerInfos.includes(info))
+		.map((info) => [info.handler, 'onExit']);
+	const chains = []
```
Comment (josex2r) [inline addon/index.js]: Apply destructuring:

```javascript
const chains = [...onExitChains, ...onEnterChains];
```

Or single concat:

```javascript
const chains = [].concat(onExitChains, onEnterChains);
```

```javascript
const chains = onExitChains.concat(onEnterChains);
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,118 @@
...
+[![NPM](https://nodei.co/npm/ember-route-promise-chain.png?downloads=true&downloadRank=true)](https://nodei.co/npm/ember-route-promise-chain/)
+
+A tiny ember-cli addon to enable promise-based hooks upon entering or exiting routes.
+
+## Usage
+
+Create a new [instance-initializer](https://guides.emberjs.com/v2.14.0/applications/initializers/) and inject the addon.
+
+```javascript
+import injectPromiseChain from 'ember-route-promise-chain';
```
[the remark this answers is not in the document]
Comment (adrigzr) [reply README.md]: I'm following the same approach as [ember-load-initializers](https://github.com/ember-cli/ember-load-initializers) addon. We should not be intrusive.

File: addon/index.js (hunk 2 above)
Comment (adrigzr) [reply addon/index.js]: Nice catch!

**Pull request merged**